Builds/CMake/CMakeFuncs.cmake

```diff
@@ -147,11 +147,7 @@ macro(setup_build_cache)
     set(CMAKE_CONFIGURATION_TYPES
       DebugClassic
       ReleaseClassic)
-    if(${CMAKE_BUILD_TYPE} STREQUAL "Debug")
-      set(CMAKE_BUILD_TYPE DebugClassic)
-    elseif(${CMAKE_BUILD_TYPE} STREQUAL "Release")
-      set(CMAKE_BUILD_TYPE ReleaseClassic)
-    endif()
+    set(CMAKE_BUILD_TYPE "${CMAKE_BUILD_TYPE}Classic")
```

Comment:
I've also noticed `-Dtarget=gcc.release` doesn't set a build type. This is a bug (though obviously not introduced in this PR).